R/display.R

```diff
@@ -2,25 +2,35 @@
 #'
 #' @param message character with any message you want
 #' @param position character with position of the parameter. Default "top right".
+#' @param type character with display type to specify the id. Default to "message"
 #'
-#' @return div which wraps your message to display it in the top-right corner of your shiny app.
+#' @return div which wraps your message to display it in the {position} corner of your shiny app.
 #' @export
 #' @import glue
-display <- function(message, position = "top right") {
+display <- function(message, position = "top right", type = "message") {
   allowed_positions <- c("top right", "top left", "bottom right", "bottom left")
   if (!position %in% allowed_positions)
     stop("Position argument not allowed.")
+  fixed_layout <- "fixed"
+  # when called from inside infoPanel function position arguments are supposed to be overwritten
+  # and css position should be empty
+  if (isTRUE(any(grepl("infoPanel", sys.call(-8))))) {
+    position <- get_args()$position
+    fixed_layout <- ""
+  }
   splitted_position <- unlist(strsplit(position, " "))
   position_vertical <- splitted_position[1] #nolint
   position_horizontal <- splitted_position[2] #nolint
+  type_id <- paste0("shinyinfo_", type)
+
   tagList( #nolint
     tags$head(
       tags$style(
         HTML(
           glue::glue(
             "
-            #shinyinfo {{
-              position: fixed;
+            #{type_id} {{
+              position: {fixed_layout};
```

Comment:
What about creating a common class for all style-related features and leave position related to these items? I think it's more elegant, but I'm happy to discuss.